File: ios/HybridRiveLogger.swift
import NitroModules

class HybridRiveLogger: HybridRiveLoggerSpec {
    func setHandler(handler: @escaping (String, String, String) -> Void) throws {
        RiveLog.handler = handler
    }

    func resetHandler() throws {
        RiveLog.handler = nil
    }

    func setLogLevel(level: String) throws {
        guard let parsed = RiveLogLevel(string: level) else {
            throw RuntimeError.error(withMessage: "Invalid log level '\(level)'. Use: debug, info, warn, error")
        }
        RiveLog.minLevel = parsed
    }
}

File: android/src/main/java/com/margelo/nitro/rive/HybridRiveLogger.kt
package com.margelo.nitro.rive

import androidx.annotation.Keep
import com.facebook.proguard.annotations.DoNotStrip

@Keep
@DoNotStrip
class HybridRiveLogger : HybridRiveLoggerSpec() {
    override fun setHandler(handler: (level: String, tag: String, message: String) -> Unit) {
        RiveLog.handler = handler
    }

    override fun resetHandler() {
        RiveLog.handler = null
    }

    override fun setLogLevel(level: String) {
        val parsed = RiveLogLevel.fromString(level)
            ?: throw RuntimeException("Invalid log level '$level'. Use: debug, info, warn, error")
        RiveLog.minLevel = parsed
    }
}
